Ideas/easier-embedded-rust.md

```diff
@@ -0,0 +1,57 @@
+
+How can this be done efficiently? Arduino and CircuitPython both have corporate backing in addition to their open source efforts. While there is a growing open source embedded rust community, any new approaches would have to leverage existing work or start from scratch.
+
+How can this be done in a way that has a "path" to more traditional Rust usage? When users become more comfortable with embedded Rust, and would like to make lower level modifications, how can they do this without abandoning everything they've learned, or without experiencing a steep learning curve?
+
+## Potential Research Areas
+
+### Better enumeration of Why and How other approaches are easier
+
+It would be good to do an overview of today's Arduino and CircuitPython environments, and see what they provide (or don't) to make the experience easier. It would also be useful to capture where their approaches "fall over", or what limitations typical users run into.
+
+### Using a small OS/Runtime
+
+This approach is somewhat similar to how CircuitPython approaches the problem: Have a small, portable runtime that provides base functionality. This core is compiled, and is often target (or CPU family) specific.
+
+This base is extended by statically or dynamically included modules. In CircuitPython this can be pre-compiled modules, or textual python code. In most or all cases, this can be done without "flashing" the device (in the traditional sense), often achieved by "copying files" to a virtual flash drive interface for portability.
+
+This approach may limit application to targets with enough resources (typically RAM and Flash, including external flash) to handle both an "OS"/"Host Environment", as well as the user application.
+
+### Provide a "common enough" base library
+
+This approach is somewhat similar to how Arduino approaches the problem: making code "seem" the same to the end user, by forcing libraries to use a common, high level set of interfaces.
+
+User code is compiled together with selected libraries to produce a single, static code image which is flashed to the device (either via debugger or bootloader).
+
+Arduino does some extra work here by providing it's own IDE and package manager ("board" and "library" managers) to manage dependencies, that may be already available out of the box in Rust.
+
+Although Rust does have language level support for standard interfaces via Traits (rather than 'common header/api' approach used in C++/Arduino) - leveraged heavily by embedded-hal, the "OSS zeitgeist" of Rust is to provide MUCH more control over implementation details that are important to "power users", but less so to "hobbyist/artist" users (at least for initial usage). This is often confusing or even overwhelming for newcomers. This is very commonly seen on "early main code" - setting up clocks, peripherals, and other resources.
```

Comment:
This first sentence is quite long, let me see if I can understand/flow it out:
- Although Rust does have language level support for standard interfaces via Traits (rather than the 'common header/api' approach used in C++/Arduino) - leveraged heavily by embedded-hal - the "OSS Zeitgeist" of Rust is to provide MUCH more control over implementation details that are important to "power users", but less so to "hobbyist/artist" users (at least for initial usage).

ok I think the only thing throwing me off a bit is the "- leveraged heavily by embedded-hal -" part, if it is supposed to be a parenthetical statement or.. well, I think this sentence could be hashed out a bit for clarity, or separated into a couple sentences.